### ISSUE TITLE
Fuzzer: multiple ls attached to type 1

### ISSUE THREAD

**Issue by elunna** [elunna/hackem#135]
```
Suddenly, the dungeon collapses.
multiple ls attached to type 1
Generating more information you may report:

[0] /lib/x86_64-linux-gnu/libasan.so.6(+0x45c0e) [0x7ff836f83c0e]
[1] ./hackemdir/hackem(+0x50bdb8) [0x5613f9e5fdb8]
[2] ./hackemdir/hackem(+0x50bca8) [0x5613f9e5fca8]
[3] ./hackemdir/hackem(panic+0x6c0) [0x5613f9e66bc7]
[4] ./hackemdir/hackem(impossible+0x2d1) [0x5613fa3c7dab]
[5] ./hackemdir/hackem(del_light_source+0x3f3) [0x5613f9f38394]
[6] ./hackemdir/hackem(rehumanize+0x5d2) [0x5613fa3d79e9]
[7] ./hackemdir/hackem(mdamageu+0x174) [0x5613fa101585]
[8] ./hackemdir/hackem(+0x78f8d0) [0x5613fa0e38d0]
[9] ./hackemdir/hackem(mattacku+0x10208) [0x5613fa0cadd5]
[10] ./hackemdir/hackem(dochug+0xe9f4) [0x5613fa26e169]
[11] ./hackemdir/hackem(dochugw+0x2b17) [0x5613fa25831b]
[12] ./hackemdir/hackem(movemon+0x185a) [0x5613fa1c4909]
[13] ./hackemdir/hackem(moveloop+0x60d) [0x5613f9b6f23e]
[14] ./hackemdir/hackem(main+0xe0a) [0x5613fa7f98ca]
[15] /lib/x86_64-linux-gnu/libc.so.6(+0x29d90) [0x7ff836a01d90]
[16] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x80) [0x7ff836a01e40]
[17] ./hackemdir/hackem(_start+0x25) [0x5613f9b6cac5]

Program received signal SIGABRT, Aborted.
__pthread_kill_implementation (no_tid=0, signo=6, threadid=140704050001856) at ./nptl/pthread_kill.c:44
44	./nptl/pthread_kill.c: No such file or directory.
(rr) p moves
$1 = 6847482
(rr) bt
#0  __pthread_kill_implementation (no_tid=0, signo=6, threadid=140704050001856) at ./nptl/pthread_kill.c:44
#1  __pthread_kill_internal (signo=6, threadid=140704050001856) at ./nptl/pthread_kill.c:78
#2  __GI___pthread_kill (threadid=140704050001856, signo=signo@entry=6) at ./nptl/pthread_kill.c:89
#3  0x00007ff836a1a476 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#4  0x00007ff836a007f3 in __GI_abort () at ./stdlib/abort.c:79
#5  0x00005613f9e5fcce in NH_abort () at end.c:236
#6  0x00005613f9e66bc7 in panic (str=0x5613fa96e280 "%s") at end.c:802
#7  0x00005613fa3c7dab in impossible (s=0x5613fa8ff5e0 "multiple ls attached to type %d") at pline.c:518
#8  0x00005613f9f38394 in del_light_source (type=1, id=0x5613fac23ca0 <tmp_anything>) at light.c:128
#9  0x00005613fa3d79e9 in rehumanize () at polyself.c:1273
#10 0x00005613fa101585 in mdamageu (mtmp=0x6110002c5dc0, n=27) at mhitu.c:4003
#11 0x00005613fa0e38d0 in hitmu (mtmp=0x6110002c5dc0, mattk=0x5613faa59da4 <mons+56708>) at mhitu.c:2746
#12 0x00005613fa0cadd5 in mattacku (mtmp=0x6110002c5dc0) at mhitu.c:1013
#13 0x00005613fa26e169 in dochug (mtmp=0x6110002c5dc0) at monmove.c:998
#14 0x00005613fa25831b in dochugw (mtmp=0x6110002c5dc0) at monmove.c:126
#15 0x00005613fa1c4909 in movemon () at mon.c:1567
#16 0x00005613f9b6f23e in moveloop (resuming=0 '\000') at allmain.c:244
#17 0x00005613fa7f98ca in main (argc=0, argv=0x7fff4f0570d8) at ../sys/unix/unixmain.c:353
(rr) bt -ful
#0  __pthread_kill_implementation (no_tid=0, signo=6, threadid=140704050001856) at ./nptl/pthread_kill.c:44
        tid = <optimized out>
        ret = 0
        pd = 0x7ff836eb37c0
        old_mask = {__val = {140704045180192, 1073741824, 0, 140704047114112, 4294967295, 140704034586624, 
            140704047097344, 16120361145486120192, 140734519138224, 140704047114112, 0, 16120361145486120192, 
            17591814892278, 140734519138224, 17591814892278, 140734519138224}}
        ret = <optimized out>
        pd = <optimized out>
        old_mask = <optimized out>
        ret = <optimized out>
        tid = <optimized out>
        ret = <optimized out>
        resultvar = <optimized out>
        resultvar = <optimized out>
        __arg3 = <optimized out>
        __arg2 = <optimized out>
        __arg1 = <optimized out>
        _a3 = <optimized out>
        _a2 = <optimized out>
        _a1 = <optimized out>
        __futex = <optimized out>
        resultvar = <optimized out>
        __arg3 = <optimized out>
        __arg2 = <optimized out>
        __arg1 = <optimized out>
        _a3 = <optimized out>
        _a2 = <optimized out>
        _a1 = <optimized out>
        __futex = <optimized out>
        __private = <optimized out>
--Type <RET> for more, q to quit, c to continue without paging--
        __oldval = <optimized out>
        result = <optimized out>
#1  __pthread_kill_internal (signo=6, threadid=140704050001856) at ./nptl/pthread_kill.c:78
No locals.
#2  __GI___pthread_kill (threadid=140704050001856, signo=signo@entry=6) at ./nptl/pthread_kill.c:89
No locals.
#3  0x00007ff836a1a476 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
        ret = <optimized out>
#4  0x00007ff836a007f3 in __GI_abort () at ./stdlib/abort.c:79
        save_stage = 1
        act = {__sigaction_handler = {sa_handler = 0x7fff4f055940, sa_sigaction = 0x7fff4f055940}, sa_mask = {
            __val = {140734519139296, 140734519139296, 140704051332287, 140704047097344, 16120361145486120192, 
              140734519138224, 140704047114112, 18446744073709547520, 140704049942528, 140704052560600, 
              140704051369645, 140734519139296, 140704051355670, 17591814892278, 140734519132160, 140704045180192}}, 
          sa_flags = 1874171136, sa_restorer = 0x7fff4f055940}
        sigs = {__val = {32, 140734519138544, 140704050854926, 94643796966840, 94643796966568, 94643796995015, 
            94643802635691, 94643797853076, 94643802700265, 94643799725445, 94643799603408, 16120361145486120192, 
            94643801219433, 140734519139072, 140734519139072, 140734519138576}}
#5  0x00005613f9e5fcce in NH_abort () at end.c:236
        gdb_prio = 1
        libc_prio = 2
        aborting = 1 '\001'
#6  0x00005613f9e66bc7 in panic (str=0x5613fa96e280 "%s") at end.c:802
        the_args = {{gp_offset = 16, fp_offset = 48, overflow_arg_area = 0x7fff4f055bd0, 
            reg_save_area = 0x7fff4f055b00}}
#7  0x00005613fa3c7dab in impossible (s=0x5613fa8ff5e0 "multiple ls attached to type %d") at pline.c:518
        pbuf = "multiple ls attached to type 1\000\000h\r\000\000\000\000\000\000 ", '\000' <repeats 15 times>, " \000\000\000\060\000\000\000\220k\005O\377\177\000\000\320j\005O\377\177\000\000\n\000\000\000\370\177\000\000\240C\210\372\023V\000\000\002\000\000\000\000\000\000\000\002\000\000\000\000\000\000\000\377\377\377\377\377\377\377\377\000\225\265o\r\a\267\337\002\000\000\000\000\000\000\000\300],\000\020a\000\000\320b\005O\377\177\000\000\256\253\340\351\377\017\000\000p]\005O\377\177\000\000p]\005O\377\177\000\000\000c\005O\377\177\000\000\221\001\325\371\023V\000\000\377\377\3--Type <RET> for more, q to quit, c to continue without paging--
77\377\377\377\377\377"...
        the_args = {{gp_offset = 16, fp_offset = 48, overflow_arg_area = 0x7fff4f0562c0, 
            reg_save_area = 0x7fff4f0561e0}}
#8  0x00005613f9f38394 in del_light_source (type=1, id=0x5613fac23ca0 <tmp_anything>) at light.c:128
        curr = 0x0
        prev = 0x0
        next = 0x0
        tmp_id = {a_void = 0x0, a_obj = 0x0, a_monst = 0x0, a_int = 0, a_char = 0 '\000', a_schar = 0 '\000', 
          a_uchar = 0 '\000', a_uint = 0, a_long = 0, a_ulong = 0, a_iptr = 0x0, a_lptr = 0x0, a_ulptr = 0x0, 
          a_uptr = 0x0, a_string = 0x0, a_nfunc = 0x0, a_mask32 = 0}
        found_it = 1 '\001'
#9  0x00005613fa3d79e9 in rehumanize () at polyself.c:1273
        was_flying = 1 '\001'
#10 0x00005613fa101585 in mdamageu (mtmp=0x6110002c5dc0, n=27) at mhitu.c:4003
No locals.
#11 0x00005613fa0e38d0 in hitmu (mtmp=0x6110002c5dc0, mattk=0x5613faa59da4 <mons+56708>) at mhitu.c:2746
        mdat = 0x5613faa59d88 <mons+56680>
        uncancelled = 1
        ptmp = 0
        dmg = 27
        armpro = 0
        permdmg = 0
        tmphp = 0
        buf = "Pf\005O\377\177\000\000\361\226)\372\023V\000\000\340e\005O\377\177\000\000`e\005O\377\177\000\000`e\005O\377\177\000\000\300],\000\020a\000\000@g\005O\377\177\000\000\000\225\265\000\003\000\000\000\377\377\377\377A\000\000\000\365\004\375\066\370\177\000\000\340e\005O\377\177", '\000' <repeats 18 times>, "%i\005O\377\177\000\000\263\212\265A\000\000\000\000\240W\223\372\023V\000\000I\212)\372\023V\000\000C\000\000\000\016\000\000\000\000\000\000\000\001\000\000\000\001\000\000\000\001\000\000\000\000\000\000\000", '\377' <repeats 12 times>, "\004\000\000\000\036", '\000' <repeats 23 times>...
        olduasmon = 0x5613faa552c8 <mons+37544>
        res = 1
--Type <RET> for more, q to quit, c to continue without paging--
        burnmsg = 0 '\000'
        armask = 393232
        hated_obj = 0x0
        vorpal_wield = 0 '\000'
#12 0x00005613fa0cadd5 in mattacku (mtmp=0x6110002c5dc0) at mhitu.c:1013
        mattk = 0x5613faa59da4 <mons+56708>
        alt_attk = {aatyp = 5 '\005', adtyp = 17 '\021', damn = 3 '\003', damd = 21 '\025'}
        i = 3
        j = 4
        k = 0
        tmp = 36
        sum = {1, 1, 1, 0, 1325755376, 32767}
        mdat = 0x5613faa59d88 <mons+56680>
        marmf = 0x0
        ranged = 0 '\000'
        range2 = 0 '\000'
        foundyou = 1 '\001'
        youseeit = 1 '\001'
        skipnonmagc = 0 '\000'
#13 0x00005613fa26e169 in dochug (mtmp=0x6110002c5dc0) at monmove.c:998
        mdat = 0x5613faa59d88 <mons+56680>
        tmp = <optimized out>
        mdummy = 0x5613fac11cb0 <level+26864>
        inrange = 1
        nearby = 1
        scared = 0
        oldx = -541653235
        oldy = 1325755568
        mwalk_sewage = 0 '\000'
#14 0x00005613fa25831b in dochugw (mtmp=0x6110002c5dc0) at monmove.c:126
        x = 76
--Type <RET> for more, q to quit, c to continue without paging--
        y = 16
        already_saw_mon = 0 '\000'
        rd = 24848
#15 0x00005613fa1c4909 in movemon () at mon.c:1567
        mtmp = 0x6110002c5dc0
        nmtmp = 0x611000463740
        somebody_can_move = 1 '\001'
#16 0x00005613f9b6f23e in moveloop (resuming=0 '\000') at allmain.c:244
        moveamt = 9
        wtcap = 0
        change = 0
        monscanmove = 0 '\000'
        elf_regen = 1 '\001'
        orc_regen = 1 '\001'
        vamp_regen = 1 '\001'
#17 0x00005613fa7f98ca in main (argc=0, argv=0x7fff4f0570d8) at ../sys/unix/unixmain.c:353
        fd = -1
        dir = 0x0
        exact_username = 0 '\000'
        resuming = 0 '\000'
        plsel_once = 1 '\001'
(rr) 
```

**Comment by elunna** [comment]
Not able to reproduce and stale

**Comment by elunna** [comment]
Popped up again, this time I'm able to see more of what's happening. A genetic engineer is definitely causing shenanigans.

```
rr) bt
#0  __pthread_kill_implementation (no_tid=0, signo=6, threadid=140500057802688) at ./nptl/pthread_kill.c:44
#1  __pthread_kill_internal (signo=6, threadid=140500057802688) at ./nptl/pthread_kill.c:78
#2  __GI___pthread_kill (threadid=140500057802688, signo=signo@entry=6) at ./nptl/pthread_kill.c:89
#3  0x00007fc8b7c1a476 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#4  0x00007fc8b7c007f3 in __GI_abort () at ./stdlib/abort.c:79
#5  0x000055eddb9dde9d in NH_abort () at end.c:236
#6  0x000055eddb9e4f16 in panic (str=0x55eddc562200 "%s") at end.c:802
#7  0x000055eddbf5ec8d in impossible (s=0x55eddc4ee2a0 "multiple ls attached to type %d") at pline.c:685
#8  0x000055eddbaba09d in del_light_source (type=1, id=0x55eddc83a700 <tmp_anything>) at light.c:128
#9  0x000055eddbf679f6 in polyself (psflags=0) at polyself.c:697
#10 0x000055eddbc1245c in mon_poly (magr=0x6110004a5f00, mdef=0x55eddc82e700 <youmonst>, dmg=1) at mhitm.c:2806
#11 0x000055eddbc6b704 in hitmu (mtmp=0x6110004a5f00, mattk=0x55eddc65bc88 <mons+52328>) at mhitu.c:2580
#12 0x000055eddbc54919 in mattacku (mtmp=0x6110004a5f00) at mhitu.c:1021
#13 0x000055eddbdfdf78 in dochug (mtmp=0x6110004a5f00) at monmove.c:1020
#14 0x000055eddbde7f0d in dochugw (mtmp=0x6110004a5f00) at monmove.c:163
#15 0x000055eddbd4fd6e in movemon () at mon.c:1573
#16 0x000055eddb6dd22c in moveloop (resuming=0 '\000') at allmain.c:247
#17 0x000055eddc3e5773 in main (argc=0, argv=0x7ffdd0170c88) at ../sys/unix/unixmain.c:353


(rr) p moves
$1 = 1845949

(rr) p u.uz
$2 = {dnum = 0 '\000', dlevel = 3 '\003'}

(rr) p u.umonster
$3 = 749 (PM_ICE_MAGE)

(rr) p u.umonnum
$4 = 749 (PM_ICE_MAGE)

(rr) p urace.malenum
$5 = 184 (PM_ORC)

What is attacking us? PM_GENETIC_ENGINEER

(rr) p toplines
$6 = "It hits!  You are subjected to a freakish metamorphosis.  You feel like a new orc!  You can see again.", '\000' <repeats 197 times>
(rr) 


turn 1845949, was 303 PM_BLACK_LIGHT
(rr) p toplines                              #
$13 = "It hits!  You are subjected to a freakish metamorphosis.", '\000' <repeats 243 times>


turn 1845947, was 205 PM_ROTHE
(rr) p toplines
$11 = "The genetic engineer hits!  You are subjected to a freakish metamorphosis.  You turn into a black light!", '\000' <repeats 195 times>
```

**Comment by elunna** [comment]
History:

(```
rr) p toplines
$6 = "It hits!  You are subjected to a freakish metamorphosis.  You feel like a new orc!  You can see again.", '\000' <repeats 197 times>
(rr) 

turn 1845949, was 303 PM_BLACK_LIGHT
(rr) p toplines                              #
$13 = "It hits!  You are subjected to a freakish metamorphosis.", '\000' <repeats 243 times>

turn 1845947, was 205 PM_ROTHE
(rr) p toplines
$11 = "The genetic engineer hits!  You are subjected to a freakish metamorphosis.  You turn into a black light!", '\000' <repeats 195 times>

turn 1845945, was 749 (PM_ICE_MAGE)
(rr) p toplines
$20 = "(2 pts.)  You hit it.  The genetic engineer hits!  You are subjected to a freakish metamorphosis.  You turn into a rothe!", '\000' <repeats 178 times>


```

**Comment by elunna** [comment]
After going back many many polymorphs in the recording, I don't think this anything to do with masks anymore. ublindf did not have anything associated with it.

**Comment by k21971** [comment]
I had this happen in an evilhack fuzz session a few days ago, in that case a wand of polymorph was used against the player by a monster. Same spot in light.c and polyself.c

**Comment by elunna** [comment]
These commits from 3.7 look promising:
https://github.com/NetHack/NetHack/commit/fdf0bfd8a4ef640fb263d4935fc6fd3a675870a1
https://github.com/NetHack/NetHack/commit/9e942262356ec258b232815ede9942eee754fc43
https://github.com/NetHack/NetHack/commit/e0e8fced7c7336f9afcb87c6c0f3e6ad628c04c4

**Comment by elunna** [comment]
Might have a fix in 46d6a70bf. Not verified yet but it looks like it might fit the bill.

**Comment by elunna** [comment]
Closing - too difficult to repro, doesn't seem like a real problem for players.
